wicker/core/datasets.py

```diff
@@ -303,3 +452,58 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+def build_dataset(
+    dataset_name: str,
+    dataset_version: str,
+    dataset_partition_name: str,
+    columns_to_load: Optional[List[str]] = None,
+    treat_objects_as_bytes: bool = False,
+    filters=None,
+) -> AbstractDataset:
+    """Builder function to make it easier for users to create instances of Wicker datasets as the
+    builder encapsulates many of the parameters necessary for the dataset creation.
+
+    This function determines the Wicker dataset type and reads out the remainder of the required
+    properties for the dataset from the Wicker config file.
+
+    :param dataset_name: name of the dataset
+    :param dataset_version: version of the dataset
+    :param dataset_partition_name: partition name
+    :param columns_to_load: list of columns to load, defaults to None which loads all columns
+    :param treat_objects_as_bytes: If set, don't try to decode ObjectFields and keep them as binary data.
+    :param filters: Only returns rows which match the filter. Defaults to None, i.e., returns all rows.
+    :type filters: pyarrow.compute.Expression, List[Tuple], or List[List[Tuple]], optional
+    .. seealso:: `filters in <https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html>`__ # noqa
+    """
+    config = get_config()
+    if config.filesystem_config.loaded:
```

Comment:
I think the type of the dataset should be one of the input parameter, instead of controlled by config. There maybe use case that need to create both S3datasets and FilesystemDatasets at the same time

Reply:
@zhenyu I was thinking the same thing. Then I started to think we might need to have multiple `WickerAwsS3Config` or `WickerFileSystemConfig` entries in the `wickerconfig.json` configuration file.

Do you think that is a possible usecase? If so, then I can add a `configuration_name` property to `WickerAwsS3Config` and `WickerFileSystemConfig` and change our new builder function so that it accepts a particular `configuration_name`.

Hower, if `wickerconfig.json` can only contain exactly one `WickerAwsS3Config` and `WickerFileSystemConfig`, then I will change the builder function to accept the type of the requested dataset.

Reply:
Do you think that is a possible usecase?  I think at least we should not block the customer to do some possible options.
Second, I think we should have multiple entries in the config file for each Naming space. We only need one S3 because s3 is a naming space, we prob need more filesystem config because each mounted Dir is a naming space. WDYT? 
I think the original wicker mix the naming space with the storage class and filesystem concept together. Let us improve it little by little

Reply:
@zhenyu Yes, I am thinking one S3Dataset but multiple FileSystemDatasets is a usecase. I will push a short update to support this.